pkg/auth/azure.go

```diff
@@ -0,0 +1,138 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2021 freiheit.com*/
+package auth
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/MicahParks/keyfunc"
+	jwt "github.com/golang-jwt/jwt/v4"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+var jwks *keyfunc.JWKS = nil
+var clientId string
+var tenantId string
+
```

Comment:
Ideally, there should be no global variables.
Could you make it so that they are passed in from outside? then server.go would keep the state.

Reply:
moved state to server.go